src/lib/pages/frontsite/SelfCheckout/SelfCheckout.js

```diff
@@ -0,0 +1,144 @@
+import React from 'react';
+
+import PropTypes from 'prop-types';
+import Overridable from 'react-overridable';
+import { Container, Header } from 'semantic-ui-react';
+import { BarcodeScanner } from '@components/BarcodeScanner';
+import { Html5Qrcode, Html5QrcodeSupportedFormats } from 'html5-qrcode';
+import { SelfCheckoutModal } from './SelfCheckoutModal';
+import { invenioConfig } from '@config';
+import _get from 'lodash/get';
+
+class SelfCheckout extends React.Component {
```

Comment:
Actually, ignore my other comments: there is a mix of responsibilities here that we need to untangle.

This cmp should only care about checking out or display messages. We should expect to have only one callback here:
```
onBarcodeScanned(barcode) {
  ... checkout ...
```

And all the rest should be in the cmp that is handing the scanner. Here, both cmps know how to use the scanner, while we really wanted to encapsulate/hide the scanner in the child cmp (or another object).
If you have to pause/resume the camera, you should pass down a prop that says `cameraHidden` `true/false` or `checkingOut` or similar (find better naming), and the child cmp should know that it has to call pause or resume.
Basically, the parent should not know much how to use the camera. It is your child BarcodeScanner that is the glue in-between the React app and the library.

Reply:
Thanks a lot for the input, having 2 components share responsibilities isn't ideal, the other aspect I kept in mind was to implement the components such that the library becomes easily replaceable later on if needed. Do you think it'll be better to have 3 components?
- Parent as it is right now, with it not knowing how to access the camera/scanner (it'll encapsulate the UI elements including SelfCheckout) [name: SelfCheckoutPage]
- Middle component which has all the controls/functionality (a generic connector between the app and a scanner library) [name: SelfCheckout]
- Child component which is where the scanner library related code goes) [name: BarcodeScanner]

Do you think that will make it complex?

Reply:
I agree with Nico on the separation of concerns, lets discuss it in person!